api_tests/nodes/views/test_node_contributors_detail.py

```diff
@@ -56,115 +56,110 @@ def project_private(self, user, title, description, category):
             creator=user
         )
 
-    @pytest.fixture()
-    def make_resource_url(self):
-        def make_resource_url(resource_id, user_id):
-            return '/{}nodes/{}/contributors/{}/'.format(
-                API_BASE, resource_id, user_id)
-        return make_resource_url
+    def make_resource_url(self, resource_id, user_id):
+        return f'/{API_BASE}nodes/{resource_id}/contributors/{user_id}/'
 
     @pytest.fixture()
     def url_public(self, user, project_public):
-        return '/{}nodes/{}/contributors/{}/'.format(
-            API_BASE, project_public._id, user._id)
+        return f'/{API_BASE}nodes/{project_public._id}/contributors/{user._id}/'
 
-    @pytest.fixture()
-    def url_private_base(self, project_private):
-        return '/{}nodes/{}/contributors/{}/'.format(
-            API_BASE, project_private._id, '{}')
+    def url_private(self, node, user_id):
+        return f'/{API_BASE}nodes/{node._id}/contributors/{user_id}/'
 
-    @pytest.fixture()
-    def url_private(self, user, url_private_base):
-        return url_private_base.format(user._id)
-
-    def test_get_contributor_detail_valid_response(
-            self, app, user, project_public,
-            project_private, url_public, url_private):
-
-        #   test_get_public_contributor_detail
+@pytest.mark.django_db
+@pytest.mark.enable_implicit_clean
+class TestContributorDetail(ContributorDetailMixin):
```

Comment:
ContributorDetailMixin had to be changed to remove the prefix Test- or else it runs independently of the subclass.